mypy/stubtest.py

```diff
@@ -116,10 +116,14 @@ def get_description(self, concise: bool = False) -> str:
             return _style(self.object_desc, bold=True) + " " + self.message
 
         stub_line = None
-        stub_file: None = None
+        stub_file = None
         if not isinstance(self.stub_object, Missing):
             stub_line = self.stub_object.line
-        # TODO: Find a way of getting the stub file
+        # TODO: find out how we can get filenames from other nodes:
+        if isinstance(self.stub_object, nodes.TypeInfo):
+            stub_file = os.path.abspath(
+                self.stub_object.module_name.replace(".", os.path.sep) + ".pyi"
```

Comment:
This might not be accurate. Sometimes stubtest's definitions come from `.py` files (e.g. with packages that have mixed inline types and stubs). While it's rare for such definitions to be involved in errors, it's possible, and it would be very confusing if we lied.

Reply:
It's also not necessarily the case that the stubs are relative to the current directory. They could be installed

Reply:
yes, this is more complex than I thought!